app/src/processing/app/Base.java

```diff
@@ -1585,6 +1640,21 @@ public void actionPerformed(ActionEvent actionevent) {
 
     JRadioButtonMenuItem item = new JRadioButtonMenuItem(action);
 
+    // create an action for the "recent boards" copy of this menu item
+    // which clicks the original menu item
+    Action actionClone = new AbstractAction(board.getName()) {
+      public void actionPerformed(ActionEvent actionevent) {
+        item.setSelected(true);
+        item.getAction().actionPerformed(new ActionEvent(this, -1, ""));
+      }
+    }; 
+
+    // create a menu item for the "recent boards" menu
+    JMenuItem itemClone = new JMenuItem(actionClone);
```

Comment:
This was my original intention but it gets a little complicated. It requires creating a `ButtonGroup` every time we update the list and adding all of the new buttons to it, which isn't wild, but it also requires that menu to "know" when a board that isn't on the recent list is selected so that it can manually unselect all boards in the recent list. I couldn't come up with an elegant way to do this on first blush but if people think it's worthwhile, I can give it another go. I just had nightmares of the radio buttons getting out of sync between the recent list and the all boards list.

Reply:
Isn't this something that `rebuildRecentBoardsList` could do? Just select the first item and deselect all others? Or more generically, select any item that matches the current item and deselect others (which should be effectively the same)?